### PR TITLE
Webapp mdc

### PR DESCRIPTION
**Pull request by odra**
## Additional Information

MDC required changes for msn and webapp usage

### Mobile Enabled

* Run an install using this pr branch
* Login into the webapp and check if mdc url is listed in the right menu
* Run the uninstall playbook

### Mobile Disabled

* Run the install playbook adding the following flags; `-e ups=false -e mdc=false -e datasync=false -e mobile_security_service=false`
* Login into the webapp and check if mdc service is not listed in the right menu
* Check if any of the mobile services were created
* Ru the uninstall playbook using the same vars listed in the first bullet item

## Is an upgrade task required and are there additional steps needed to test this?

- [ ] Tested Installation
- [ ] Tested Uninstallation
- [ ] Tested or Created follow on for Upgrade

### CODE REVIEW

**Comment by trepel** [comment]
Running on my PDS and PoC cluster

File: roles/3scale/tasks/users.yml [hunk 1]
```diff
@@ -47,6 +47,15 @@
...
 
+- name: Wait for system-app rollout
+  shell: "oc rollout status dc/system-app -n {{ threescale_namespace }} --watch=false | grep 'rolled out'"
+  register: result
+  until: result.stdout
+  retries: 50
+  delay: 10
+  failed_when: result.stderr
+  changed_when: False
+
```
Comment (trepel) [inline roles/3scale/tasks/users.yml]: @odra I am not sure if this fixes INTLY-2871. In order to fix INTLY-2871 this piece of code needs to be added before the `Redeploy system-app` step. That is not the best place though, the best would be to place this piece of code just after the piece of code that causing the system-app redeploy (but I don't know where that is).

File: playbooks/install_mobile_services.yml [hunk 2]
```diff
@@ -41,4 +41,15 @@
         name: rhsso-user
       when:
         - user_rhsso | default(true) | bool
-        - mdc | default(true) | bool
\ No newline at end of file
+        - mdc | default(true) | bool
+    
+    - name: Install managed services broker
```
Comment (wei-lee) [inline playbooks/install_mobile_services.yml]: I thought the default installer will install msb, no? Why do we need to install it again?

File: playbooks/install_services.yml [hunk 3]
```diff
@@ -137,12 +137,21 @@
...
     - name: Expose vars
       include_vars: "../roles/apicurito/defaults/main.yml"
+    
+    - name: Set additional walkthrough paths
+      block:
+        - set_fact:
+            webapp_wt_list: []
+        - set_fact:
+            webapp_wt_list: "{{ webapp_wt_list + [mobile_walkthrough_location] }}"
+          when: mdc
```
Comment (wei-lee) [inline playbooks/install_services.yml]: is the indentation right here for the `when` statement?

**Comment by wei-lee** [comment]
left a few comments, otherwise LGTM. I think @grdryn is trying it atm. Will let him approve the PR once it's verified.

File: playbooks/install_mobile_services.yml (hunk 2 above)
Comment (odra) [reply playbooks/install_mobile_services.yml]: because it needs to re-apply the msb config for mdc

File: playbooks/install_services.yml (hunk 3 above)
Comment (odra) [reply playbooks/install_services.yml]: yes, it just executes the mdc stuff in here but I encapsulated all additional wt tasks in a block

File: roles/3scale/tasks/users.yml (hunk 1 above)
Comment (odra) [reply roles/3scale/tasks/users.yml]: it is added just after the rollout, this command checks fo the rollout status

Comment (odra) [reply roles/3scale/tasks/users.yml]: I got what you mean, I think the config map changes is triggering a new deployment so I moved it before the rollout command

**Pull request merged**